src/wallet/wallet2.cpp

```diff
@@ -1573,20 +1575,27 @@ bool wallet2::should_expand(const cryptonote::subaddress_index &index) const
 //----------------------------------------------------------------------------------------------------
 void wallet2::expand_subaddresses(const cryptonote::subaddress_index& index)
 {
+  using spair_t = std::pair<crypto::public_key, cryptonote::subaddress_index>;
+
   hw::device &hwdev = m_account.get_device();
+  std::vector<std::pair<crypto::public_key, cryptonote::subaddress_index>> new_elements;
   if (m_subaddress_labels.size() <= index.major)
   {
     // add new accounts
-    cryptonote::subaddress_index index2;
     const uint32_t major_end = get_subaddress_clamped_sum(index.major, m_subaddress_lookahead_major);
+    size_t count = 0;
+    for (size_t i = m_subaddress_labels.size(); i < major_end; ++i)
+      count += get_subaddress_clamped_sum((i == index.major ? index.minor : 0), m_subaddress_lookahead_minor);
```

Comment:
NIT: Prefer STL algorithms over raw-loops (std::accumulate in the numeric header).